gr-blocks/lib/moving_average_impl.cc

```diff
@@ -44,11 +46,11 @@ moving_average_impl<T>::moving_average_impl(int length,
       d_updated(false)
 {
     this->set_history(length);
-    // we don't have C++11's <array>, so initialize the stored vector instead
     // we store this vector so that work() doesn't spend its time allocating and freeing
     // vector storage
     if (d_vlen > 1) {
-        d_sum = std::vector<T>(d_vlen);
+        d_sum = volk::vector<T>(d_vlen);
+        d_scales = volk::vector<T>(d_vlen, d_scale);
```

Comment:
Is this really required? `volk_32f_s32f_normalize()` or `volk_32f_s32f_multiply_32f()` can be used instead of `volk_32f_x2_multiply_32f()`, even for the `gr_complex` case (apply type cast and double the vector length).

[1] https://www.libvolk.org/doxygen/volk_32fc_s32fc_multiply_32fc.html
[2] https://www.libvolk.org/doxygen/volk_32f_s32f_normalize.html

Reply:
For `gr_complex` that assumes the scaling factor is not complex, no?

Reply:
That's correct. A moving average with a complex scaling factor would be odd though. I'm fine with it either way.